stl/debugger/STL.natvis

```diff
@@ -1955,4 +2043,149 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+      <Item Name="Width index">_Specs._Fill_length</Item>
+      <Item Name="Precision index">_Specs._Fill</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::basic_format_parse_context&lt;*&gt;">
+    <DisplayString>{_Format_string}</DisplayString>
+    <Expand>
+      <Item Name="Arguments">_Num_args</Item>
+      <Item Name="Next arg id">_Next_arg_id</Item>
+    </Expand>
+  </Type>
+
+  <Type Name="std::basic_format_args&lt;*&gt;">
+    <DisplayString>Num args:{_Num_args}</DisplayString>
+    <Expand>
+      <Item Name="Type">_Index_array->_Type_</Item>
+      <IndexListItems>
+        <Size>_Num_args</Size>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 1">
+          *(reinterpret_cast&lt;const int*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 2">
+          *(reinterpret_cast&lt;const unsigned int*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 3">
+          *(reinterpret_cast&lt;const long long*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 4">
+          *(reinterpret_cast&lt;const unsigned long long*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 5">
+          *(reinterpret_cast&lt;const bool*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 6">
+          *(reinterpret_cast&lt;const unsigned char*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 7">
+          *(reinterpret_cast&lt;const float*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 8">
+          *(reinterpret_cast&lt;const double*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 9">
+          *(reinterpret_cast&lt;const long double*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 10">
+          (void*)(*(reinterpret_cast&lt;const unsigned char*&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index))
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 11">
+          reinterpret_cast&lt;char*&gt;(*((reinterpret_cast&lt;const long long *&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)+0))
+        </ValueNode>
+        <ValueNode Condition="_Index_array[$i]._Type_ == 12">
+          reinterpret_cast&lt;char*&gt;(*((reinterpret_cast&lt;const long long *&gt;(_Index_array + _Num_args) + _Index_array[$i]._Index)+0))

```

Comment:
Is this correct for `_CString_type` and `_String_type`? Again I haven't tested this, but this is using a slightly modified form of the "read a `long long`" expression, then reinterpreting it as `char*`. I don't think that'll work for 32-bit architectures.

Additionally, if my understanding is correct, for `_CString_type` we store a pointer (expecting it to be to null-terminated data) while for `_String_type` we store a `basic_string_view` (handling potentially non-null-terminated data), so I would expect to see different logic for them.